src/lib/basic/FontParams.svelte

```diff
@@ -0,0 +1,120 @@
+<script lang="ts">
+  import { createEventDispatcher, onMount } from "svelte";
+  import { tr } from "../../utils/i18n";
+  import MdIcon from "./MdIcon.svelte";
+
+  type ChangeValue = { family: string | undefined; size: number | undefined };
+
+  export let hasFamily: boolean = true;
+  export let hasSize: boolean = true;
+  export let family: string = '';
+  export let size: number = 0;
+  export let sizeMin: number = 1;
+  export let sizeMax: number = 999;
+  export let sizeStep: number = 2;
+
+  const dispatch = createEventDispatcher<{
+    change: { newValue: ChangeValue };
+  }>();
+
+  let familys: Record<string, string> = {};
+
+  const fontQuerySupported = typeof queryLocalFonts !== "undefined";
+  async function refreshFonts() {
+    if (fontQuerySupported) {
+      window.fontCache = await queryLocalFonts();
```

Comment:
Use svelte stores for global data

https://github.com/MultiMote/niimblue/blob/20c1e6c012cfd3032b7fce193e672a368ad0caf1/src/stores.ts#L20